src/pages/Cart/Cart.js

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import './Cart.scss';
+
+const Cart = () => {
+  const navigate = useNavigate();
+  const [cartItems, setCartItems] = useState([]);
+  const [selected, setSelected] = useState([]);
+
+  useEffect(() => {
+    fetch('http://localhost:3000/data/data.json')
+      .then(res => res.json())
+      .then(data => setCartItems(data));
+  }, []);
+
+  const handleCheck = (checked, id) => {
+    if (checked) {
+      setSelected(prev => [...prev, id]);
+    } else {
+      setSelected(selected.filter(ele => ele !== id));
+    }
+  };
+
+  const handleAllCheck = () => {
+    if (isAllChecked) {
+      const idArr = [];
+      cartItems.forEach(ele => idArr.push(ele.id));
+      setSelected(idArr);
```

Comment:
조건에 따른 분기처리가 반대가 된 것 같습니다.
그리고 이 부분은 이렇게 처리해볼 수 있겠네요!
```suggestion
      setSelected(cartItems.map(({ id }) => id));
```